File: Source/MKL/examples/vmlf/source/vsasinh.f
	programm MKL_VML_TEST

	include "func_interv.fi"

	real*4 sA(VEC_LEN)
	real*4 sB1(VEC_LEN)
	integer i

	MaxRMS=0.0
	do i=1,VEC_LEN
		sA(i)=VML_SASINH_BEG+((VML_SASINH_END-VML_SASINH_BEG)*i)/VEC_LEN
		sB1(i)=0;
	end do

	call VSASINH(VEC_LEN,sA,sB1)

	print *,"vsAsinh test/example programm"
	print *,""
	print *,""
	print *,"           Argument                     vsAsinh"
	print *,"=============================================================================="
	do i=1,VEC_LEN
		print 10,sA(i),sB1(i)
	end do

10	format(F25.14,F25.14)

	end

File: Source/MKL/examples/vmlc/source/vsasinh.c
#include <stdio.h>
#include <math.h>
#include "func_interv.h"

#include "mkl_vml.h"

int main()
{
  float fA[VEC_LEN],fB1[VEC_LEN];

  int i=0,vec_len=VEC_LEN;

  for(i=0;i<vec_len;i++) {
    fA[i]=(float)(__VML_SASINH_BEG+((__VML_SASINH_END-__VML_SASINH_BEG)*i)/vec_len);
    fB1[i]=0;
  }

#if defined(MKL_VML_MIXED)
  vsAsinh(vec_len,fA,fB1);
#elif defined(MKL_VML_LOWER)
  vsasinh(&vec_len,fA,fB1);
#elif defined(MKL_VML_UPPER)
  VSASINH(&vec_len,fA,fB1);
#else
# error You must define MKL_VML_CDECL for CDECL call convention or MKL_VML_STDCALL for STDCALL call convention.
#endif


  printf("vsAsinh test/example programm\n\n");
  printf("           Argument                     vsAsinh\n");
  printf("===============================================================================\n");
  for(i=0;i<vec_len;i++) {
    printf("% 25.14f % 25.14f\n",fA[i],fB1[i]);
  }
}
